$OpenBSD: patch-Makefile,v 1.10 2020/12/14 23:26:59 danj Exp $

- BUILD: makefile: Update feature flags for OpenBSD (90c9d78)
- BUILD: makefile: enable crypt(3) for OpenBSD (496374e)

Index: Makefile
--- Makefile.orig
+++ Makefile
@@ -362,10 +362,11 @@ ifeq ($(TARGET),osx)
   EXPORT_SYMBOL  = -export_dynamic
 endif
 
-# OpenBSD 5.7 and above
+# OpenBSD 6.3 and above
 ifeq ($(TARGET),openbsd)
   set_target_defaults = $(call default_opts, \
-    USE_POLL USE_TPROXY USE_THREAD USE_KQUEUE USE_ACCEPT4)
+    USE_POLL USE_TPROXY USE_LIBCRYPT USE_THREAD USE_KQUEUE USE_ACCEPT4        \
+    USE_CLOSEFROM USE_GETADDRINFO)
 endif
 
 # NetBSD
@@ -455,7 +456,9 @@ BUILD_FEATURES := $(foreach opt,$(patsubst USE_%,%,$(u
 OPTIONS_CFLAGS += $(foreach opt,$(use_opts),$(if $($(opt)),-D$(opt),))
 
 ifneq ($(USE_LIBCRYPT),)
+ifneq ($(TARGET),openbsd)
 OPTIONS_LDFLAGS += -lcrypt
+endif
 endif
 
 ifneq ($(USE_SLZ),)
